hazelcast-client-legacy/src/test/java/com/hazelcast/client/listeners/AbstractListenersOnReconnectTest.java

```diff
@@ -72,7 +73,9 @@ public void run() throws Exception {
         assertTrueEventually(new AssertTask() {
             @Override
             public void run() throws Exception {
-                assertEquals(EVENT_COUNT, eventCount.get());
+                int current = eventCount.get();
+                Logger.getLogger(this.getClass()).info("[testListenersInternal] Current event count is " + current);
+                assertEquals(EVENT_COUNT, current);
```

Comment:
@sancar @cangencer Can pointed out a good point that the time for test and assertTrueEventually are both 120 secs, hence the JUnit timeout will always kick in before assertTrueEventually timeout. The issue log also verifies this. The exception in the issue is reported to be "org.junit.runners.model.TestTimedOutException: test timed out after 120000 milliseconds". 

Hence, if we make the logging as Sancar suggested, then we should play with the test timeout so that it is guaranteed that the assertTrueEventually timeout is caught. Otherwise, we can not log using the message in  assertTrueEventually.